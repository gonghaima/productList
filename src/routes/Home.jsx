import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import config from 'config';
import { login } from 'actions/index';

import { Container, utils } from 'styled-minimal';
import Background from 'components/Background';
import Logo from 'components/Logo';

const { spacer } = utils;

const HomeContainer = styled(Container)`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  min-height: 100vh;
`;

const Header = styled.div`
  margin-bottom: ${spacer(3)};
  text-align: center;

  svg {
    width: 20rem;
  }
`;

const Heading = styled.h1`
  color: #fff;
  font-size: 3.5rem;
  line-height: 1.4;
  margin-bottom: ${spacer(3)};
  margin-top: 0;
  text-align: center;

  /* stylelint-disable */
  ${utils.responsive({
    lg: `
      font-size: 4rem;
    `,
  })};
  /* stylelint-enable */
`;

export class Home extends React.PureComponent {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    user: PropTypes.object.isRequired,
  };

  handleClickLogin = () => {
    const { dispatch } = this.props;

    dispatch(login());
  };

  render() {
    return (
      <Background key="Home" data-testid="HomeWrapper">
        <HomeContainer verticlPadding>
          <Header>
            <Logo />
          </Header>
          <Heading>{config.description}</Heading>
        </HomeContainer>
      </Background>
    );
  }
}

/* istanbul ignore next */
function mapStateToProps(state) {
  return { user: state.user };
}

export default connect(mapStateToProps)(Home);
