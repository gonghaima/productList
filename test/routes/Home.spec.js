import React from 'react';

import Home from 'routes/Home';

function setup() {
  const props = {
    dispatch: () => {},
    location: {},
  };

  return shallow(<Home {...props} />);
}

describe('Private', () => {
  const wrapper = setup();

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
