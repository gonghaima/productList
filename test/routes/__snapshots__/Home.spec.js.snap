// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render properly 1`] = `
.c0 {
  background: #000 linear-gradient(to bottom,#00b4d5 0%,#002529 100%) fixed;
  color: #fff;
  min-height: 100vh;
  overflow: hidden;
  position: relative;
}

.c0:before {
  background: linear-gradient(to bottom,#000,#fff);
  bottom: 0;
  content: '';
  left: 0;
  opacity: 0.4;
  position: absolute;
  right: 0;
  top: 0;
  -webkit-transform: rotate(-20deg) scale(1.5) translate(0,43%);
  -ms-transform: rotate(-20deg) scale(1.5) translate(0,43%);
  transform: rotate(-20deg) scale(1.5) translate(0,43%);
}

.c3 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 0;
}

.c3 svg {
  height: auto;
  max-height: 100%;
  width: 4.8rem;
}

.c1 {
  box-sizing: border-box;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  max-width: 1440px;
  position: relative;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 100vh;
}

.c2 {
  margin-bottom: 16px;
  text-align: center;
}

.c2 svg {
  width: 20rem;
}

.c4 {
  color: #fff;
  font-size: 3.5rem;
  line-height: 1.4;
  margin-bottom: 16px;
  margin-top: 0;
  text-align: center;
}

@media (min-width:768px) {
  .c1 {
    padding-left: 32px;
    padding-right: 32px;
  }
}

@media (min-width:1024px) {
  .c4 {
    font-size: 4rem;
  }
}

<Background
  data-testid="HomeWrapper"
  key="Home"
>
  <div
    className="c0"
    data-testid="HomeWrapper"
  >
    <HomeContainer
      verticalPadding={false}
      verticlPadding={true}
    >
      <div
        className="c1"
      >
        <Header>
          <div
            className="c2"
          >
            <Logo>
              <div
                className="c3"
              >
                <SVGMock />
              </div>
            </Logo>
          </div>
        </Header>
        <Heading>
          <h1
            className="c4"
          >
            React-Redux-Saga Boilerplate
          </h1>
        </Heading>
      </div>
    </HomeContainer>
  </div>
</Background>
`;
